.github/workflows/release.yml

```diff
@@ -1,58 +1,113 @@
-name: Release conda-store-ui
+name: "Build and maybe release conda-store-ui"
 
 on:
-  # we want to make a release whenever a new tag is created
+  release:
+    types: [published]
   push:
-    tags:
-      - "*"
+    branches: [main]
+    tags: ["*"]
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
 
-jobs:
-  call-build:
-    uses: conda-incubator/conda-store-ui/.github/workflows/build.yml@main
+env:
+  FORCE_COLOR: "1"
+  PACKAGE_FILE: "conda-store-ui.tgz"
 
-  make-release:
+jobs:
+  # always build and verify
+  build-application:
+    name: "Build conda-store-ui"
     runs-on: ubuntu-latest
-    # ensure that the artifacts are available from the build job
-    needs: call-build
 
     steps:
       - name: "Checkout repository 🛎"
         uses: actions/checkout@v4
 
-      # Setup .npmrc file to publish to npm
       - name: "Set up Node.js 🧶"
         uses: actions/setup-node@v4
         with:
-          node-version: 18
-          registry-url: "https://registry.npmjs.org"
-          scope: "@conda-store-ui"
+          node-version: "20.x"
+          cache: "yarn"
 
       - name: "Install dependencies 📦"
         run: yarn
 
-      - name: "Lint code 🔎"
+      - name: "Lint code 🔍"
         run: yarn eslint:check
 
-      - name: "Download webpack bundle 📦"
+      - name: "Build application 🏗"
+        run: yarn run build
+
+      - name: "Build application artifacts 🏗"
+        # ensure we use the prod target to minimise assets
+        run: yarn run webpack:prod bundle
+
+      - name: "Generate package tarball 📦"
+        run: yarn pack --filename ${{ env.PACKAGE_FILE }}
+
+      - name: "Upload package tarball 📤"
+        uses: actions/upload-artifact@v4
+        with:
+          name: conda-store-ui-package
+          path: ${{ env.PACKAGE_FILE }}
+
+  release-to-npmjs:
```

Comment:
I thought we only want to run this job on a release but I can't seem to find anything that prevents the `release-to-npm` job from running on every pull request or push to the main branch.

Reply:
We have `if: github.repository_owner == 'conda-incubator' && github.event_name == 'release' && startsWith(github.ref, 'refs/tags/')` which serves that purpose on the attestation and release

Reply:
we want the build and verification to run at all the times and the publish to npmjs only on releases

Reply:
I suppose that is when the logs are helpful. 
I mean, I can further separate the release and have it in a completely isolated job it really makes no difference IMO

Reply:
Yes and we do have much better logs thanks to you. :)

But it's still bad UI, but also maybe not high priority so... just gotta live with it